employee/employee-cinvoke.c

```diff
@@ -0,0 +1,84 @@
+/* This program  atomically unseal a pair of capabilities
+ * (code and data) by calling `ldpblr`, i.e.
+ * "Load Pair of capabilities and Branch with Link".
+ * See "ARM Architecture Reference Manual Supplement Morello"
+ * (https://developer.arm.com/documentation/ddi0606/latest).
+ */
+
+#include "../include/common.h"
+#include <assert.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/sysctl.h>
+
+struct cheri_object
+{
+	void *__capability codecap;
+	void *__capability datacap;
```

Comment:
What happens if you swap these over? I'm still looking, but in the Morello pseudo-code it looks like the data capability comes first.

Reply:
Right. I'll swap them over.

Reply:
In 51e24fd, after the swap you have suggested, the target is called (so we can reach `print_details`) but it fails with `CHERI protection violation`.